File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let completed_runner = {};
    
    for(let runner of completion) {
        if(!(!!completed_runner[runner])) {
            completed_runner[runner] = 0;
        }
        completed_runner[runner]++;
    }
    
    for(let runner of participant) {
        if(!(!!completed_runner[runner])) {
            return runner;
        }
        if(completed_runner[runner] <= 0) {
            return runner;
        }
        completed_runner[runner]--;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    challenger = {}
    done = {}
    
    for player in participant:
        if not challenger.get(player):
            challenger[player] = 0
        challenger[player] += 1
        
    for player in completion:
        if not done.get(player):
            done[player] = 0
        done[player] += 1
        
    for player in challenger:
        if not done.get(player) or done[player] != challenger[player]:
            return player
